File: Permutation.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> arr(2*n);
        for (auto &x:arr)
            cin >> x;
        vector<int> used(n), ans;
        for (int i=0; i<2*n; i++) {
            if (!(used[arr[i]-1])) {
                used[arr[i]-1] = true;
                ans.push_back(arr[i]);
            }
        }
        for (auto &x:ans)
            cout << x << " ";
        cout << endl;
    }

    return 0;
}

File: Python/Permutation.py
def main():
    t = int(input())
    while t:
        n = int(input())
        arr = [int(input()) for i in range(2*n)]
        ans = []
        for i in arr:
            if i not in ans:
                ans.append(i)
                print(i)
        t -= 1


if __name__ == '__main__':
    main()
